signer/src/transaction_coordinator.rs

```diff
@@ -170,13 +191,13 @@ where
     #[tracing::instrument(skip_all, name = "tx-coordinator")]
     pub async fn run(mut self) -> Result<(), Error> {
         tracing::info!("starting transaction coordinator event loop");
-        let mut signal_stream = self.context.as_signal_stream(&self.network);
+        let mut signal_stream = self.context.as_signal_stream(run_loop_message_filter);
 
         loop {
             match signal_stream.next().await {
-                Some(Ok(SignerSignal::Command(SignerCommand::Shutdown))) => break,
-                Some(Ok(SignerSignal::Command(SignerCommand::P2PPublish(_)))) => {}
-                Some(Ok(SignerSignal::Event(event))) => {
+                Some(SignerSignal::Command(SignerCommand::Shutdown)) => break,
+                Some(SignerSignal::Command(SignerCommand::P2PPublish(_))) => {}
```

Comment:
nit: this case should be impossible given the filter above, should we add a catchall instead?

Reply:
yeah, the main difference is that you can see which commands you're skipping here at first glance, and know that you aren't skipping something important. Then again, we have a catch all for the events. If we had only three events I would replace the catchall with the explicit variant.